clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2184,9 +2178,138 @@ void handleFailure(Throwable throwable) {
+        }
+
+        // First, we need to retrieve the node info.
+        DescribeClusterResult clusterResult = describeCluster();
+        Map<Integer, Node> nodes;
+        try {
+            nodes = clusterResult.nodes().get().stream().collect(Collectors.toMap(Node::id, node -> node));
+        } catch (InterruptedException | ExecutionException e) {
+            completeAllExceptionally(topicFutures.values(), e);
+            return new HashMap<>(topicFutures);
         }
+
+        final long now = time.milliseconds();
+        Call call = new Call("describeTopicPartitions", calcDeadlineMs(now, options.timeoutMs()),
+            new LeastLoadedNodeProvider()) {
+            Map<String, TopicRequest> pendingTopics =
+                topicNamesList.stream().map(topicName -> new TopicRequest().setName(topicName))
+                    .collect(Collectors.toMap(topicRequest -> topicRequest.name(), topicRequest -> topicRequest, (t1, t2) -> t1, TreeMap::new));
+
+            DescribeTopicPartitionsRequestData.Cursor requestCursor = null;
+            TopicDescription partiallyFinishedTopicDescription = null;
+
+            @Override
+            DescribeTopicPartitionsRequest.Builder createRequest(int timeoutMs) {
+                DescribeTopicPartitionsRequestData request = new DescribeTopicPartitionsRequestData()
+                    .setTopics(new ArrayList<>(pendingTopics.values()))
+                    .setResponsePartitionLimit(options.partitionSizeLimitPerResponse());
+                request.setCursor(requestCursor);
+                return new DescribeTopicPartitionsRequest.Builder(request);
+            }
+
+            @Override
+            void handleResponse(AbstractResponse abstractResponse) {
+                DescribeTopicPartitionsResponse response = (DescribeTopicPartitionsResponse) abstractResponse;
+                DescribeTopicPartitionsResponseData.Cursor responseCursor = response.data().nextCursor();
+
+                for (DescribeTopicPartitionsResponseTopic topic : response.data().topics()) {
+                    String topicName = topic.name();
+                    Errors error = Errors.forCode(topic.errorCode());
+
+                    KafkaFutureImpl<TopicDescription> future = topicFutures.get(topicName);
+                    if (error != Errors.NONE) {
+                        future.completeExceptionally(error.exception());
+                        topicFutures.remove(topicName);
+                        pendingTopics.remove(topicName);
+                        if (responseCursor != null && responseCursor.topicName().equals(topicName)) {
+                            responseCursor = null;
+                        }
+                        continue;
+                    }
+
+                    TopicDescription currentTopicDescription = getTopicDescriptionFromDescribeTopicsResponseTopic(topic, nodes);
+
+                    if (requestCursor != null && requestCursor.topicName().equals(topicName)) {
```

Comment:
Because the current implementation sends the topic list with the cursor, if the cursor topic is missing from the topic list, the request will fail with the invalid request error.

Reply:
But the user may not have specified the topic list at all, is it a good experience to show them an error if a topic they didn't specify got deleted?

Reply:
In any case, I think it would be more robust if we didn't rely on the previous cursor, so that if we eventually change the code to work without topic lookup we don't have subtle bugs.